src/firestore/delete.ts

```diff
@@ -0,0 +1,486 @@
+"use strict";
+
+import * as clc from "cli-color";
+import * as ProgressBar from "progress";
+
+import * as api from "../api";
+import * as firestore from "../gcp/firestore";
+import { FirebaseError } from "../error";
+import * as logger from "../logger";
+import * as utils from "../utils";
+
+// Datastore allowed numeric IDs where Firestore only allows strings. Numeric IDs are
+// exposed to Firestore as __idNUM__, so this is the lowest possible negative numeric
+// value expressed in that format.
+const MIN_ID = "__id-9223372036854775808__";
+
+export class FirestoreDelete {
+  /**
+   * Progress bar shared by the class.
+   */
+  static progressBar = new ProgressBar("Deleted :current docs (:rate docs/s)\n", {
```

Comment:
I'm a little confused why this is a static on the class, to be honest. Consider when improving the logic to remove it or make it a local variable where it's used...